### PR TITLE
fix: json schema 403 error

### PR DESCRIPTION
**Pull request by jeromevdl**
**Issue #, if available:** #1455

## Description of changes:

Use `addUriTranslator` to use local files instead of remote and thus avoid http 403 error

**Checklist**

 

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [x] Update tests
* [ ] Update docs
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java [hunk 1]
```diff
@@ -257,7 +259,10 @@ private static JsonSchema createJsonSchema(String schema) {
             String filePath = schema.substring(CLASSPATH.length());
             try (InputStream schemaStream = ValidationAspect.class.getResourceAsStream(filePath)) {
 
-                jsonSchema = ValidationConfig.get().getFactory().getSchema(schemaStream);
+                SchemaValidatorsConfig config = new SchemaValidatorsConfig();
+                config.addUriTranslator(URITranslator.prefix("https://json-schema.org", "resource:"));
```
[the remark this answers is not in the document]
Comment (jeromevdl) [reply powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java:263]: they are all (4,6,7,2019,2020) packaged locally (in the dependency jar itself, that's also why I removed in our resources)

[the remark this answers is not in the document]
Comment (scottgerring) [reply powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java:263]: sweet!

[the remark this answers is not in the document]
Comment (scottgerring) [reply powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java:263]: Weird this isn't the default configuration?

File: powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java [hunk 2]
```diff
@@ -270,13 +275,13 @@ private static JsonSchema createJsonSchema(String schema) {
     }
 
     private static void validateSchema(String schema, JsonSchema jsonSchema) {
-        String version = ValidationConfig.get().getSchemaVersion().toString();
+        String schemaId = ValidationConfig.get().getSchemaVersion().getId().replace("https://json-schema.org", "");
```
Comment (scottgerring) [inline powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java:278]: What about schemas that aren't on json-schema.org?

Comment (jeromevdl) [reply powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java:278]: there are different level of schemas: 
- the schema that the user defines (to check the api gw input for example)
- the schema of the schema itself (meta schema)

We support the meta schemas 4, 6, 7, 2019, 2020, which are all on json-schema.org. And we support user schemas which are based on these meta-schemas.

there are 2 kind of validations here: 
- validating the input (using the user-defined schema)
- validating the schema itself (using the meta schema available locally in the json-schema-validator.jar or remotely on https://json-schema.org)

Comment (scottgerring) [reply powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/ValidationUtils.java:278]: Sounds good. This is the level of comprehension I didn't have time to achieve :)

**Review by scottgerring — APPROVED** [review]

**Pull request merged**